sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorAsyncClient.java

```diff
@@ -0,0 +1,140 @@
+    private final String consumerGroupName;
+    private final EventPosition initialEventPosition;
+    private final BiFunction<PartitionContext, CheckpointManager, Subscriber<EventData>> partitionProcessorFactory;
+    private final PartitionManager partitionManager;
+    private final Map<String, EventHubConsumer> partitionConsumers = new ConcurrentHashMap<>();
+    private final String instanceId;
+    private final String eventHubName;
+    private Disposable disposable;
+    private Scheduler scheduler;
+
+    /**
+     * Package-private constructor. Use {@link EventHubClientBuilder} to create an instance
+     */
+    EventProcessorAsyncClient(EventHubAsyncClient eventHubAsyncClient, String consumerGroupName,
+        BiFunction<PartitionContext, CheckpointManager, Subscriber<EventData>> partitionProcessorFactory,
+        EventPosition initialEventPosition, PartitionManager partitionManager, String eventHubName) {
+        this.eventHubAsyncClient = Objects.requireNonNull(eventHubAsyncClient, "eventHubAsyncClient cannot be null");
+        this.consumerGroupName = Objects.requireNonNull(consumerGroupName, "consumerGroupname cannot be null");
+        this.partitionProcessorFactory = Objects.requireNonNull(partitionProcessorFactory, "partitionProcessorFactory cannot be null");
+        this.partitionManager = Objects.requireNonNull(partitionManager, "partitionManager cannot be null");
+        this.initialEventPosition = Objects
+            .requireNonNull(initialEventPosition, "initialEventPosition cannot be null");
+        this.instanceId = UUID.randomUUID().toString();
+        this.eventHubName = eventHubName;
+    }
+
+    /**
+     * Starts the event processor
+     */
+    public void start() {
+
+        scheduler = Schedulers.newElastic("EventProcessorAsyncClient");
+        disposable = scheduler.schedule(this::run);
+    }
+
+    /**
+     * Stops the event processor
+     */
+    public void stop() {
+        this.partitionConsumers.forEach((key, value) -> {
+            try {
+                logger.info("Closing event hub consumer for partition {}", key);
+                value.close();
+                partitionConsumers.remove(key);
+            } catch (IOException ex) {
+                logger.warning("Unable to close event hub consumer for partition {}", key);
+            }
+        });
+        disposable.dispose();
+        scheduler.dispose();
+        logger.info("Stopping the event processor");
+    }
+
+    /* Internal implementation. This is only a simple demo to show how it may look
+    - will need to move it to a different class and add more details to this */
+    void run() {
+        /* This should run periodically get new ownership details and close/open new
+        consumers when ownership of this instance has changed */
+
+        List<String> eventHubPartitionIds = new ArrayList<>();
+        this.eventHubAsyncClient.getPartitionIds().doOnNext(eventHubPartitionIds::add).blockLast();
```

Comment:
Since this is a demo, I won't comment too much... but:
1. remove `.blockLast()`
1. `doOnNext` I see for side effects like logging, but since following code depends on it, have you considered using .map and .flatMap?